sql/5_0_2-to-6_0_0_upgrade.sql

```diff
@@ -2302,6 +2302,7 @@ CREATE TABLE `oauth_clients` (
 `revoke_date` datetime DEFAULT NULL,
 `contacts` text,
 `redirect_uri` text,
+`launch_uri` text,
```

Comment:
this should also be added at the bottom of this script (to add it to folks whom are using development version of OpenEMR; ie. already have the table installed; in fact, you don't even need it here really, just need the below at the bottom of this script):
```
#IfMissingColumn oauth_clients launch_uri
ALTER TABLE `oauth_clients` ADD `launch_uri` text;
#EndIf
```

Reply:
@adunsulag I thought I already had this covered with `initiate_login_uri` which is used for supporting smart ehr app launch.

Reply:
Well that's funny, I must have missed that.  I probably wasn't thinking that launching the app had anything to do with logging in so my eyes must have glazed over it haha.

Reply:
I sneak up on folks in that way:) I don't know if I document poorly because I don't want to be responsible or if i'm just lazy. I'll leave that to the reader to decide:)

I think I need to add the user role to the trusted user depending on how user signed in to the server(OpenEMR or Patient). This will be the role to base permissions on. 
Currently the client_role determines the client sign in method i.e code flow with secret or code flow with PKCE(public app).
However, user will sign in as user or patient thus that is the role/context we need for permissions. Meaning that will help qualify what scopes is allowed in scope validation. Stephen, I think this will help with your launch scopes issue.

Reply:
@sjpadgett Do you save off the user_id as part of the trusted_user, the user_role could be derived from the user_id, the fact that they are separate tables would make it easier if we saved off the role as well, but we could work around it if we have the user_id saved.

Reply:
I do save the user_id as UUID but then determining role from it is kind of a chore. The main help for user_role in trusted user is so we can pass that role to the scope validator w/o a bunch of conditionals to determine validators role.
I could add it with a PR i'm doing to validate requested scopes in dynamic register.

Reply:
I think it would be possible for a client to be confidential and still have a patient role/context that allows limited write.
i.e the client role only determines authorization and not authentication.

Am I over thinking again?

Reply:
In the standalone patient launch you have confidential apps that will be restricted to only patient role.  My understanding is that any request for 'patient/*' permission will have those resources be restricted to just a single patient context ie limited read/write.

Reply:
That's true, however; we determine what scopes the patient role is allowed when we authenticate by what the client role is set to and not the user. Having trusted user role set offers an easier way to validate scopes by role and serve as an additional check on resource server for those trying to pass in unauthorized scopes.

Also, When I authenticate user, I'm relying on token scope validation to catch unauthorized scopes from role after user has authorized their scopes and claims. i.e I'm just passing forward what was requested using the client role instead of authenticating users role.

Maybe i'm not making sense as many engineers don't have the ability to read minds however, I think I just talked myself into having to admit, i've made a boo boo in my login design. We'll keep that between us though, right!

Reply:
Fixed this so we don't deal with launch_uri now, it uses the initiate_login_uri